File: eCinema.Model/Film.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCinema.Model
{
    public class Film
    {
        public int FilmId { get; set; }
        public string Naziv { get; set; }
        public string Sadrzaj { get; set; }
        public string? Slika { get; set; }
        public int VrijemeTrajanje { get; set; }
        public string Redatelj { get; set; }
        public string Godina { get; set; }
        public DateTime PocetakPrikazivanja { get; set; }

        public override string ToString()
        {
            return Naziv;
        }
    }
}

File: ecinema_mobile/lib/models/film.dart
import 'package:json_annotation/json_annotation.dart';

part 'film.g.dart';

@JsonSerializable()
class Film {
  int filmId;
  String naziv;
  String sadrzaj;
  String? slika;
  int vrijemeTrajanje;
  String redatelj;
  String godina;
  DateTime pocetakPrikazivanja;

  Film({
    required this.filmId,
    required this.naziv,
    required this.sadrzaj,
    this.slika,
    required this.vrijemeTrajanje,
    required this.redatelj,
    required this.godina,
    required this.pocetakPrikazivanja,
  });

  factory Film.fromJson(Map<String, dynamic> json) => _$FilmFromJson(json);
  Map<String, dynamic> toJson() => _$FilmToJson(this);
}
